apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -288,3 +296,58 @@ export const forgotPassword = async (req: Request, res: Response) => {
     })
   }
 }
+
+export const resetPassword = async (req: Request, res: Response) => {
+  try {
+    const validResetBody = resetPasswordSchema.strict().safeParse(req.body)
+
+    if (!validResetBody.success) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: 'Invalid request body.',
+        errors: validResetBody.error,
+      })
+    }
+
+    const { newPassword, email } = validResetBody.data
+    //Let's extract user ID (longstring) from the URL parameter & verify it
+    const { verify } = req.query
+    if (!verify || typeof verify !== 'string') {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: 'Invalid or missing verification token.',
+      })
+    }
+
+    const resetToken = await UserResetPasswordToken.findOne({
+      where: { id: verify },
+    })
+    //Lets check if token is found,valid & unexpired
+    if (!resetToken || !Jwt.isTokenExpired(resetToken.token)) {
+      return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).json({
+        message: 'Invalid or expired verification token.',
+      })
+    }
+
+    const user = await Users.findOne({ where: { email } })
+    //Lets handle when a user is not found
+    if (!user) {
+      return res.status(HTTP_STATUS_CODE.NOT_FOUND).json({
+        message: 'User not found.',
+      })
+    }
+
+    const hashedPassword = await PasswordHarsher.hash(newPassword)
+    //Lets update the password
+    await Users.update({ password: hashedPassword }, { where: { email } })
+    //Let's invalidate the reset token to prevent its reuse
+    await resetToken.destroy()
+
+    return res.status(HTTP_STATUS_CODE.SUCCESS).json({
+      message: [`Password reset successful.`],
+    })
```

Comment:
Let's discuss this part of your code

Reply:
Ok

Reply:
Please when can we have this discussion? I made updates to the code based on your review recommendations though.